### PR TITLE
feat(ormuse): Development environment

### PR DESCRIPTION
**Pull request by mohsenHa**
### Description

This pull request introduces Docker Compose configuration files for essential services required by the project, organized into separate YAML files for improved modularity and clarity:

- `go.yml`: Configures the main application containers (at this time only destination) using the Go image with version tag 1.22 and can configured in `.env` file
- `rabbitmq.yml`: Defines the RabbitMQ service container configuration.
- `redis.yml`: Specifies the Redis service container configuration.
- `traefik.yml`: Sets up the Traefik service container configuration.

### Changes Made:
Created separate Docker Compose YAML files (`go.yml`, `rabbitmq.yml`, `redis.yml`, `traefik.yml`) for each service.
Defined service-specific configurations within each YAML file.
Add two files `docker-compose.bash` and `docker-compose.bat` for easy use docker-compose.

### Development Environment Enhancements:

- Configured the local development environment to leverage these Docker containers for service isolation and easy setup.

This modular approach enhances the maintainability of the Docker Compose configurations and provides a clear structure for managing different services within the project.

### CODE REVIEW

**Comment by iam-benyamin** [comment]
Hey Mohsen, I was delighted to see the PR commit

File: serve/docker-compose/services/go.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      - ../../../:/home/www
+    depends_on:
+      - redis
+      - rabbitmq
+    environment:
+      ORMUS_DESTINATION__RABBITMQ_CONSUMER_CONNECTION__HOST: rabbitmq
+      ORMUS_DESTINATION__RABBITMQ_TASK_MANAGER_CONNECTION__HOST: rabbitmq
+      ORMUS_DESTINATION__REDIS_TASK_IDEMPOTENCY__HOST: redis
+      ORMUS_REDIS__HOST: redis
+#    labels:
```
Comment (iam-benyamin) [inline serve/docker-compose/services/go.yml:27]: please remove dead code

**Review by iam-benyamin — APPROVED** [review]
great Job!

**Comment by gocastsian** [comment]
Please Document the usage of `traefik` for local development.

**Comment by mohsenHa** [comment]
I was eagerly anticipating this morning's meeting but couldn't find the link; I thought it was canceled.

As outlined in the documentation at [how_use_docker_compose.md](https://github.com/mohsenHa/ormus/blob/main/doc/how_use_docker_compose.md), I integrated Traefik to expose the RabbitMQ manager service on the domain rabbitmq.ormuse.local. In the future, we can employ it for other services like the source, where we'll need to expose the service.

However, for the time being, we can omit it and directly expose the RabbitMQ manager port on localhost.
@gocastsian

**Review by gocastsian — APPROVED** [review]

**Pull request merged**